In [134]:
import pandas as pd

In [135]:
employee_dataset = pd.read_excel("general_data.xlsx",sheet_name=0)

In [136]:
employee_dataset.isnull()

Age  Attrition  BusinessTravel  Department  DistanceFromHome  \
0     False      False           False       False             False   
1     False      False           False       False             False   
2     False      False           False       False             False   
3     False      False           False       False             False   
4     False      False           False       False             False   
...     ...        ...             ...         ...               ...   
4405  False      False           False       False             False   
4406  False      False           False       False             False   
4407  False      False           False       False             False   
4408  False      False           False       False             False   
4409  False      False           False       False             False   

      Education  EducationField  EmployeeCount  EmployeeID  Gender  ...  \
0         False           False          False       False   False  ...   
1         False           False          False       False   False  ...   
2         False           False          False       False   False  ...   
3         False           False          False       False   False  ...   
4         False           False          False       False   False  ...   
...         ...             ...            ...         ...     ...  ...   
4405      False           False          False       False   False  ...   
4406      False           False          False       False   False  ...   
4407      False           False          False       False   False  ...   
4408      False           False          False       False   False  ...   
4409      False           False          False       False   False  ...   

      NumCompaniesWorked  Over18  PercentSalaryHike  StandardHours  \
0                  False   False              False          False   
1                  False   False              False          False   
2                  False   False              False          False   
3                  False   False              False          False   
4                  False   False              False          False   
...                  ...     ...                ...            ...   
4405               False   False              False          False   
4406               False   False              False          False   
4407               False   False              False          False   
4408               False   False              False          False   
4409               False   False              False          False   

      StockOptionLevel  TotalWorkingYears  TrainingTimesLastYear  \
0                False              False                  False   
1                False              False                  False   
2                False              False                  False   
3                False              False                  False   
4                False              False                  False   
...                ...                ...                    ...   
4405             False              False                  False   
4406             False              False                  False   
4407             False              False                  False   
4408             False              False                  False   
4409             False               True                  False   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0              False                    False                 False  
1              False                    False                 False  
2              False                    False                 False  
3              False                    False                 False  
4              False                    False                 False  
...              ...                      ...                   ...  
4405           False                    False                 False  
4406           False    

In [137]:
employee_dataset.drop_duplicates()

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   
4409   40        No      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   
4409    Male  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [138]:
employee_dataset.dropna()

Age Attrition     BusinessTravel              Department  \
0      51        No      Travel_Rarely                   Sales   
1      31       Yes  Travel_Frequently  Research & Development   
2      32        No  Travel_Frequently  Research & Development   
3      38        No         Non-Travel  Research & Development   
4      32        No      Travel_Rarely  Research & Development   
...   ...       ...                ...                     ...   
4404   29        No      Travel_Rarely                   Sales   
4405   42        No      Travel_Rarely  Research & Development   
4406   29        No      Travel_Rarely  Research & Development   
4407   25        No      Travel_Rarely  Research & Development   
4408   42        No      Travel_Rarely                   Sales   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4404                 4          3          Other              1        4405   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4404  Female  ...                 1.0      Y                18              8   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4404                 0               6.0                      2   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...                   ..

In [139]:
employee_dataset.dtypes

Age                          int64
Attrition                   object
BusinessTravel              object
Department                  object
DistanceFromHome             int64
Education                    int64
EducationField              object
EmployeeCount                int64
EmployeeID                   int64
Gender                      object
JobLevel                     int64
JobRole                     object
MaritalStatus               object
MonthlyIncome                int64
NumCompaniesWorked         float64
Over18                      object
PercentSalaryHike            int64
StandardHours                int64
StockOptionLevel             int64
TotalWorkingYears          float64
TrainingTimesLastYear        int64
YearsAtCompany               int64
YearsSinceLastPromotion      int64
YearsWithCurrManager         int64
dtype: object

In [140]:
employee_dataset.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [141]:
from sklearn import preprocessing

In [142]:
label_encoder = preprocessing.LabelBinarizer()

In [143]:
employee_dataset["Attrition"] = label_encoder.fit_transform(employee_dataset["Attrition"]) 

In [144]:
employee_dataset.dtypes

Age                          int64
Attrition                    int32
BusinessTravel              object
Department                  object
DistanceFromHome             int64
Education                    int64
EducationField              object
EmployeeCount                int64
EmployeeID                   int64
Gender                      object
JobLevel                     int64
JobRole                     object
MaritalStatus               object
MonthlyIncome                int64
NumCompaniesWorked         float64
Over18                      object
PercentSalaryHike            int64
StandardHours                int64
StockOptionLevel             int64
TotalWorkingYears          float64
TrainingTimesLastYear        int64
YearsAtCompany               int64
YearsSinceLastPromotion      int64
YearsWithCurrManager         int64
dtype: object

In [145]:
employee_dataset

Age  Attrition     BusinessTravel              Department  \
0      51          0      Travel_Rarely                   Sales   
1      31          1  Travel_Frequently  Research & Development   
2      32          0  Travel_Frequently  Research & Development   
3      38          0         Non-Travel  Research & Development   
4      32          0      Travel_Rarely  Research & Development   
...   ...        ...                ...                     ...   
4405   42          0      Travel_Rarely  Research & Development   
4406   29          0      Travel_Rarely  Research & Development   
4407   25          0      Travel_Rarely  Research & Development   
4408   42          0      Travel_Rarely                   Sales   
4409   40          0      Travel_Rarely  Research & Development   

      DistanceFromHome  Education EducationField  EmployeeCount  EmployeeID  \
0                    6          2  Life Sciences              1           1   
1                   10          1  Life Sciences              1           2   
2                   17          4          Other              1           3   
3                    2          5  Life Sciences              1           4   
4                   10          1        Medical              1           5   
...                ...        ...            ...            ...         ...   
4405                 5          4        Medical              1        4406   
4406                 2          4        Medical              1        4407   
4407                25          2  Life Sciences              1        4408   
4408                18          2        Medical              1        4409   
4409                28          3        Medical              1        4410   

      Gender  ...  NumCompaniesWorked Over18 PercentSalaryHike  StandardHours  \
0     Female  ...                 1.0      Y                11              8   
1     Female  ...                 0.0      Y                23              8   
2       Male  ...                 1.0      Y                15              8   
3       Male  ...                 3.0      Y                11              8   
4       Male  ...                 4.0      Y                12              8   
...      ...  ...                 ...    ...               ...            ...   
4405  Female  ...                 3.0      Y                17              8   
4406    Male  ...                 2.0      Y                15              8   
4407    Male  ...                 0.0      Y                20              8   
4408    Male  ...                 0.0      Y                14              8   
4409    Male  ...                 0.0      Y                12              8   

      StockOptionLevel TotalWorkingYears  TrainingTimesLastYear  \
0                    0               1.0                      6   
1                    1               6.0                      3   
2                    3               5.0                      2   
3                    3              13.0                      5   
4                    2               9.0                      2   
...                ...               ...                    ...   
4405                 1              10.0                      5   
4406                 0              10.0                      2   
4407                 0               5.0                      4   
4408                 1              10.0                      2   
4409                 0               NaN                      6   

      YearsAtCompany  YearsSinceLastPromotion  YearsWithCurrManager  
0                  1                        0                     0  
1                  5                        1                     4  
2                  5                        0                     3  
3                  8                        7                     5  
4                  6                        0                     4  
...              ...                      ...         

In [146]:
from scipy.stats import pearsonr 

In [147]:
# 1 Correlation between Attrition and Age:

stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['Age'])

print(stats,p)

if(p<0.05):
    print("There is significant correlation between Attrition and Age. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and Age. Accepting Null Hypothesis")

#

-0.15920500686577957 1.996801615887198e-26
There is significant correlation between Attrition and Age. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis


### Correlation between Attrition and Age:

 Here,
    r = -0.15920500686577957 
    
    So,  there is low negative correlation between Attrition and Age 
    
    p = 1.996801615887198e-26 
        
    P value is < 0.05, Rejecting H0 and Accepting H1, hence there is significant correlation between Attrition & Age     

In [148]:
# 2 Correlation between Attrition and DistanceFromHome:

stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['DistanceFromHome'])

print(stats,p)
if(p<0.05):
    print("There is significant correlation between Attrition and DistanceFromHome. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and DistanceFromHome. Accepting Null Hypothesis")


-0.009730141010179664 0.5182860428050771
There is no significant correlation between Attrition and DistanceFromHome. Accepting Null Hypothesis


### Correlation between Attrition and DistanceFromHome:

 Here,
 
    r = -0.009730141010179664  
    
    So,  there is low negative correlation between Attrition and DistanceFromHome 
    
    p = 0.5182860428050771 
        
    P value is > 0.05, Accepting H0, hence there is no significant correlation between Attrition & DistanceFromHome     

In [149]:
# 3 Correlation between Attrition and MonthlyIncome:

stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['MonthlyIncome'])
print(stats,p)

if(p<0.05):
    print("There is significant correlation between Attrition and MonthlyIncome. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and MonthlyIncome. Accepting Null Hypothesis")


-0.03117628169811503 0.03842748490600132
There is significant correlation between Attrition and MonthlyIncome. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis


### Correlation between Attrition and MonthlyIncome:

 Here,
 
    r = -0.03117628169811503   
    
    So,  there is low negative correlation between Attrition and MonthlyIncome 
    
    p = 0.03842748490600132
        
    P value is < 0.05, Rejecting H0 and Accepting H1 hence there is significant correlation between Attrition & MonthlyIncome     

In [150]:
# 4 Correlation between Attrition and Education:

stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['Education'])
print(stats,p)

if(p<0.05):
    print("There is significant correlation between Attrition and Education. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and Education. Accepting Null Hypothesis")


-0.015111167710968732 0.3157293177118575
There is no significant correlation between Attrition and Education. Accepting Null Hypothesis


#### Correlation between Attrition and Education:
Here,

    r = -0.015111167710968732  

    So,  there is low negative correlation between Attrition and Education 

    p = 0.3157293177118575

    P value is > 0.05, Accepting H0 hence there is no significant correlation between Attrition & Education  

In [151]:
# 5 Correlation between Attrition and JobLevel:

stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['JobLevel'])
print(stats,p)

if(p<0.05):
    print("There is significant correlation between Attrition and JobLevel. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and JobLevel. Accepting Null Hypothesis")


-0.010289713287495013 0.49451717271828405
There is no significant correlation between Attrition and JobLevel. Accepting Null Hypothesis


#### Correlation between Attrition and JobLevel:
Here,

    r = -0.010289713287495013   

    So,  there is low negative correlation between Attrition and JobLevel 

    p = 0.49451717271828405

    P value is > 0.05, Accepting H0, hence there is no significant correlation between Attrition & JobLevel  

In [152]:
# 6 Correlation between Attrition and PercentSalaryHike:

stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['PercentSalaryHike'])
print(stats,p)

if(p<0.05):
    print("There is significant correlation between Attrition and PercentSalaryHike. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and PercentSalaryHike. Accepting Null Hypothesis")


0.032532594891053486 0.030743386433355353
There is significant correlation between Attrition and PercentSalaryHike. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis


#### Correlation between Attrition and PercentSalaryHike:
Here,

    r = 0.032532594891053486   

    So,  there is positive correlation between Attrition and PercentSalaryHike 

    p = 0.030743386433355353

    P value is < 0.05, Rejecting H0 and Accepting H1, hence there is significant correlation between Attrition & PercentSalaryHike  

In [153]:
# 7 Correlation between Attrition and TrainingTimesLastYear:
stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['TrainingTimesLastYear'])
print(stats,p)

if(p<0.05):
    print("There is significant correlation between Attrition and TrainingTimesLastYear. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and TrainingTimesLastYear. Accepting Null Hypothesis")


-0.049430576244255 0.0010247061915365072
There is significant correlation between Attrition and TrainingTimesLastYear. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis


#### Correlation between Attrition and TrainingTimesLastYear:
Here,

    r = -0.049430576244255

    So,  there is negatively correlation between Attrition and TrainingTimesLastYear 

    p = 0.0010247061915365072

    P value is < 0.05, Rejecting H0 and Accepting H1, hence there is significant correlation between Attrition & TrainingTimesLastYear  

In [154]:
# 8 Correlation between Attrition and YearsWithCurrManager:
stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['YearsWithCurrManager'])
print(stats,p)

if(p<0.05):
    print("There is significant correlation between Attrition and YearsWithCurrManager. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and YearsWithCurrManager. Accepting Null Hypothesis")


-0.15619931590162872 1.7339322652896276e-25
There is significant correlation between Attrition and YearsWithCurrManager. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis


#### Correlation between Attrition and TrainingTimesLastYear:
Here,

r = -0.15619931590162872 

So,  there is negatively correlation between Attrition and TrainingTimesLastYear 

p = 1.7339322652896276e-25

P value is < 0.05, Rejecting H0 and Accepting H1, hence there is significant correlation between Attrition & TrainingTimesLastYear 

In [155]:
# 9 Correlation between Attrition and StockOptionLevel:
stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['StockOptionLevel'])
print(stats,p)

if(p<0.05):
    print("There is significant correlation between Attrition and StockOptionLevel. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and StockOptionLevel. Accepting Null Hypothesis")


-0.006838852403261531 0.6498072937475723
There is no significant correlation between Attrition and StockOptionLevel. Accepting Null Hypothesis


#### Correlation between Attrition and StockOptionLevel:
Here,

r = -0.006838852403261531  

So, there is negatively correlation between Attrition and StockOptionLevel

p =  0.6498072937475723

P value is > 0.05, Accepting H0, hence there is no significant correlation between Attrition & StockOptionLevel

In [156]:
# 10 Correlation between Attrition and YearsSinceLastPromotion:
stats,p = pearsonr(employee_dataset['Attrition'],employee_dataset['YearsSinceLastPromotion'])
print(stats,p)

if(p<0.05):
    print("There is significant correlation between Attrition and YearsSinceLastPromotion. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis")
else:
    print("There is no significant correlation between Attrition and YearsSinceLastPromotion. Accepting Null Hypothesis")


-0.03301877514258432 0.028330336189396753
There is significant correlation between Attrition and YearsSinceLastPromotion. Rejecting Null Hypothesis and Accepting the Alternate Hypothesis


#### Correlation between Attrition and YearsSinceLastPromotion:
Here,

r = -0.03301877514258432 

So, there is negatively correlation between Attrition and YearsSinceLastPromotion

p = 0.028330336189396753

P value is > 0.05, Rejecting H0 and Accepting H1, hence there is significant correlation between Attrition & YearsSinceLastPromotion